spec/actions/collection-spec.js

```diff
@@ -10,62 +10,84 @@ describe('collection action creators', () => {
-      deferred.reject();
+      it('rejects when jqXHR fails', function (done) {
+        const deferred = $.Deferred(); // eslint-disable-line new-cap
+        spyOn(this.storageManager.storage('posts'), 'fetch').and.returnValue(deferred);
+        this.store.dispatch(this.fetch('posts')).catch(done);
+        deferred.reject();
+      });
     });
-  });
 
-  describe('previous XHRs', () => {
-    it('uses a trackKey to abort previous pending XHR', function () {
-      const abort = jasmine.createSpy('abort');
-      const state = () => 'pending';
-      spyOn(this.storageManager.storage('posts'), 'fetch').and.returnValue({ abort, state });
-      this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
-      this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
+    describe('previous XHRs', () => {
+      it('uses a trackKey to abort previous pending XHR', function () {
+        const abort = jasmine.createSpy('abort');
+        const state = () => 'pending';
+        spyOn(this.storageManager.storage('posts'), 'fetch').and.returnValue({ abort, state });
+        this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
+        this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
 
-      expect(abort).toHaveBeenCalledTimes(1);
+        expect(abort).toHaveBeenCalledTimes(1);
+      });
+
+      it('does not abort non-pending previous XHR', function () {
+        const abort = jasmine.createSpy('abort');
+        const state = () => 'resolved';
+        spyOn(this.storageManager.storage('posts'), 'fetch').and.returnValue({ abort, state });
+        this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
+        this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
+
+        expect(abort).toHaveBeenCalledTimes(0);
+      });
     });
+  });
 
-    it('does not abort non-pending previous XHR', function () {
-      const abort = jasmine.createSpy('abort');
-      const state = () => 'resolved';
-      spyOn(this.storageManager.storage('posts'), 'fetch').and.returnValue({ abort, state });
-      this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
-      this.store.dispatch(this.fetch('posts', { trackKey: 'foo' })).catch(() => {});
+  describe('when an adapter is passed in', () => {
+    it('uses the adapter', function () {
+      const deferred = $.Deferred(); // eslint-disable-line new-cap
+      const stubAdapter = {
+        fetchCollection: jasmine.createSpy('fetchCollection').and.returnValue(deferred),
+      };
+      const fetchOptions = { filters: { foo: 'bar' } };
+      this.store.dispatch(this.fetch('posts', {
+        fetchOptions,
+        adapter: stubAdapter,
+      }));
 
-      expect(abort).toHaveBeenCalledTimes(0);
+      expect(stubAdapter.fetchCollection).toHaveBeenCalledWith('posts', {
+        dispatch: jasmine.any(Function),
```

Comment:
If anyone looking this code over has a good idea for testing that correct dispatch and getState were passed into this and every other action, feel free to suggest it. This is my solution for now since it turns out the dispatch passed in from the param of a called thunk is not the same as the dispatch that is evoked on line 81 above.

Reply:
Yeah, I think we're fine for now. Appreciate the comment, it will be good for historical context if it is ever dug up.